www-client/links/patches/links-2.20.2.patchset

```diff
@@ -0,0 +1,276 @@
++	char *result = cast_uchar mem_alloc(B_FILE_NAME_LENGTH);
++	if (find_directory(B_USER_SETTINGS_DIRECTORY, -1, 1, result, B_FILE_NAME_LENGTH) != B_OK) {
++		mem_free(result);
++		return NULL;
++	}
++	return result;
++#else
++	return NULL;
++#endif
++}
++
+ #ifndef BEOS
+ 
+ void block_stdin(void) {}
+@@ -2106,7 +2130,7 @@ void unblock_stdin(void) {}
+ 
+ #if defined(BEOS)
+ 
+-#include <be/kernel/OS.h>
++#include <kernel/OS.h>
+ 
+ static int thr_sem_init = 0;
+ static sem_id thr_sem;
+@@ -2907,6 +2931,13 @@ int get_system_env(void)
+ 	return 0;
+ }
+ 
++#elif defined(HAIKU)
++
++int get_system_env(void)
++{
++	return ENV_BE;
++}
++
+ #elif defined(WIN)
+ 
+ int get_system_env(void)
+@@ -3036,6 +3067,14 @@ static int open_in_new_be(struct terminal *term, unsigned char *exe, unsigned ch
+ }
+ #endif
+ 
++#ifdef HAIKU
++static int open_in_new_haiku(struct terminal *term, unsigned char *exe, unsigned char *param)
++{
++	exec_new_links(term, cast_uchar "Terminal", exe, param);
++	return 0;
++}
++#endif
++
+ #ifdef G
+ static int open_in_new_g(struct terminal *term, unsigned char *exe, unsigned char *param)
+ {
+@@ -3084,6 +3123,9 @@ static const struct {
+ #if defined(BEOS) || defined(HAIKU)
+ 	{ENV_BE, open_in_new_be, TEXT_(T_BEOS_TERMINAL), TEXT_(T_HK_BEOS_TERMINAL)},
+ #endif
++#ifdef HAIKU
++	{ENV_BE, open_in_new_haiku, TEXT_(T_BEOS_TERMINAL), TEXT_(T_HK_BEOS_TERMINAL)},
++#endif
+ #ifdef G
+ 	{ENV_G, open_in_new_g, TEXT_(T_WINDOW), TEXT_(T_HK_WINDOW)},
+ #endif
+@@ -3131,6 +3173,9 @@ int can_open_os_shell(int environment)
+ #ifdef BEOS
+ 	if (!(environment & ENV_BE)) return 0;
+ #endif
++#ifdef HAIKU
++	if (!(environment & ENV_BE)) return 0;
++#endif
+ #ifdef G
+ 	if (F && drv->flags & GD_NO_OS_SHELL) return 0;
+ #endif
+diff --git a/os_depx.h b/os_depx.h
+index c8f9db2..a31448c 100644
+--- a/os_depx.h
++++ b/os_depx.h
+@@ -152,6 +152,10 @@ int bounced_write(int fd, const void *buf, size_t size);
+ #endif
+ #endif
+ 
++#ifdef HAIKU
++/* nothing so far */
++#endif
++
+ #ifdef OPENVMS
+ #if defined(__INITIAL_POINTER_SIZE)
+ #if __INITIAL_POINTER_SIZE == 64
+@@ -368,7 +372,7 @@ typedef const char *const_char_ptr;
+ #if !defined(EXEC_IN_THREADS) || !defined(HAVE_GETHOSTBYNAME) || defined(USE_GETADDRINFO)
+ #define THREAD_SAFE_LOOKUP
+ #endif
+-
++/*
+ #if defined(DOS)
+ #define loop_select(a, b, c, d, e) dos_select(a, b, c, d, e, 1)
+ #elif defined(GRDRV_SVGALIB)
+@@ -376,8 +380,11 @@ typedef const char *const_char_ptr;
+ int vga_select(int n, fd_set *r, fd_set *w, fd_set *e, struct timeval *t);
+ #elif defined(GRDRV_ATHEOS)
+ #define loop_select ath_select
+-int ath_select(int n, fd_set *r, fd_set *w, fd_set *e, struct timeval *t);
+-#else
++int ath_select(int n, fd_set *r, fd_set *w, fd_set *e, struct timeval *t); 
++#elif defined(GRDRV_HAIKU)
++#define loop_select be_loop_select
++int be_loop_select(int n, fd_set *r, fd_set *w, fd_set *e, struct timeval *t);
++#else */
+ #define loop_select select
+-#endif
++/* #endif */
+ 
+-- 
+2.23.0
+
```

Comment:
Shouldn't this patch set be merged as part of the upstreaming effort done by @Begasus? :/

Reply:
Is this patch needed for mouse to work correctly? From the code I can see that it shouldn't cause any impact.

Reply:
@alaviss - Does your keyboard and mouse work now?